### PR TITLE
Mon 117

### PR DESCRIPTION
**Pull request by pcatalini**

### CODE REVIEW

**Review by Haugum — APPROVED** [review]

File: ns1/resource_record.go [hunk 1]
```diff
@@ -338,6 +338,22 @@ func resourceDataToRecord(r *dns.Record, d *schema.ResourceData) error {
...
+					for _, sub := range subdivisions {
+						subp := strings.Split(sub, "-")
+						if len(subp) != 2 {
+							return fmt.Errorf("invalid subidivision format. expecting (\"Country-Subdivision\") got %s", sub)
+						}
+						subdvisionsMap[subp[0]] = []string{}
+					}
+					for _, sub := range subdivisions {
+						sub := strings.Split(sub, "-")
+						subdvisionsMap[sub[0]] = append(subdvisionsMap[sub[0]].([]string), sub[1])
```
Comment (mburtless) [inline ns1/resource_record.go:353]: I might be missing something, but I don't think I understand why we need to perform a 2nd iteration through this list and split again to perform this step.  Why not just move this to the end of the previous loop?

```
	for _, sub := range subdivisions {
		subp := strings.Split(sub, "-")
		if len(subp) != 2 {
			return fmt.Errorf("invalid subidivision format. expecting (\"Country-Subdivision\") got %s", sub)
		}
		if subdvisionsMap[subp[0]] == nil {
			subdvisionsMap[subp[0]] = []string{}
		}
		subdvisionsMap[subp[0]] = append(subdvisionsMap[subp[0]].([]string), subp[1])
	}
```

**Review by mburtless — CHANGES_REQUESTED** [review]
We should really have an acceptance test for this `resource_record_test.go`.  I'd recommend extending the existing `TestAccRecord_meta` case.

File: ns1/resource_record.go (hunk 1 above)
Comment (pcatalini) [reply ns1/resource_record.go:353]: @mburtless you're right, thanks

**Review by mburtless — CHANGES_REQUESTED** [review]
@pcatalini - Thanks for adding acceptance tests.

Looks like there are still some formatting issues to clear up before we can merge:
```
gofmt needs running on the following files:
./ns1/resource_record.go
```

**Pull request merged**